### PR TITLE
Update release docs

### LINKED ISSUE #3978: Document "How to Release Docs" in the Contributor's Guide
**Describe the change you'd like to see**
As a follow on to https://github.com/knative/docs/issues/3912, document how to Release Docs in the Contributor's Guide. 

**Additional context**
https://hackmd.io/WWcVF60dT_m4Clv26v8V4w?both

### PR DESCRIPTION
**Pull request by evankanderson**
Fixes #3978

## Proposed Changes 

- Update the release docs. Since the new docs build is more automated and I expect the build script to change, I explained the high-level process, but not a line-by-line update.

### CODE REVIEW

File: help/maintainer/docs-release-process.md [hunk 1]
```diff
@@ -5,136 +5,41 @@ weight: 75
 type: "docs"
 ---
```
Comment (abrennan89) [inline help/maintainer/docs-release-process.md:6]: I think the metadata can be removed now FYI?

**Review by abrennan89 — COMMENTED** [review]
/lgtm
/hold
LGTMing this but ya'll can remove the hold once someone with more knowledge checks it (maybe @julz or @RichieEscarez )

File: help/maintainer/docs-release-process.md [hunk 2]
```diff
@@ -5,136 +5,41 @@ weight: 75
...
+window](https://github.com/knative/community/blob/main/mechanics/RELEASE-VERSIONING-PRINCIPLES.md#knative-community-support-window-principle),
+so clean up old versions at the end of the list when there will be more than 4.
 
-## Test the content
-* Update your local versions of the build config files to run a local build.
-* Run a local build using from the /website directory: scripts/localbuild.sh. Follow the extensive instructions at the top of the script file.
-* Check to make sure everything looks okay.
-* Sync your changes to the staging branch with git pull --rebase upstream main and git push upstream staging. {need more detail}
+Until the old hugo configuration has been aged out (after release 0.27), it's
+best to send these PRs to @julz to double-check. You'll also want to remove the
```
Comment (julz) [inline help/maintainer/docs-release-process.md:42]: happily there's only one more release before this happens... and I'm one of the release leads for it, so this clause may 🤞 never have to be used, other than via internal monologue.

**Review by julz — APPROVED** [review]
Very pleasing amount of red in this update 😊.

/lgtm
/approve
/unhold

**Pull request merged**